app/views/directives/create-secret.html

```diff
@@ -0,0 +1,300 @@
+            in use. Please choose a different name.</span>
+      </div>
+      <div class="help-block">
+        Unique name of the new secret.
+      </div>
+  </div>
+  <div class="form-group">
+    <label for="authentificationType">Authetication Type</label>
+    <ui-select required ng-model="newSecret.authType" search-enabled="false">
+      <ui-select-match>{{$select.selected.label}}</ui-select-match>
+      <ui-select-choices repeat="type.id as type in secretAuthTypeMap[newSecret.type].authTypes">
+        <div ng-bind-html="type.label"></div>
+      </ui-select-choices>
+    </ui-select>
+  </div>
+
+  <div ng-if="newSecret.authType === 'kubernetes.io/basic-auth'">
+    <div class="form-group">
+      <label for="username">Username</label>
+      <div>
+        <input class="form-control"
+          id="username"
+          name="username"
+          ng-model="newSecret.data.username"
+          type="text"
+          autocorrect="off"
+          autocapitalize="off"
+          spellcheck="false">
+      </div>
+      <div class="help-block">
+        Optional username for SCM servers authentication.
+      </div>
+    </div>
+
+    <div class="form-group">
+      <label for="passwordToken" class="required">Password or Token</label>
+      <div>
+        <input class="form-control"
+          id="passwordToken"
+          name="passwordToken"
+          ng-model="newSecret.data.passwordToken"
+          type="text"
+          autocorrect="off"
+          autocapitalize="off"
+          spellcheck="false"
+          required>
+      </div>
+      <div class="help-block">
+        Password or token for SCM servers authentication.
+      </div>
+    </div>
+  </div>
+
+  <div ng-if="newSecret.authType === 'kubernetes.io/ssh-auth'">
+    <div class="form-group" id="private-key">
+      <label for="privateKey" class="required">SSH Private Key</label>
+      <osc-file-input 
+        id="private-key-file-input"
+        model="newSecret.data.privateKey"
+        drop-zone-id="private-key"
+        dragging="false"
+        help-text="Upload your private SSH key file."
+        show-values="false"></osc-file-input>
+      <div ui-ace="{
+        theme: 'eclipse',
+        onLoad: aceLoaded,
+        rendererOptions: {
+          fadeFoldWidgets: true,
+          showPrintMargin: false 
+        }
+      }" ng-model="newSecret.data.privateKey" class="create-secret-editor ace-bordered" id="private-key-editor" required></div>
+      <div class="help-block">
+        Content of private SSH key file for SCM servers authentication.
+      </div>
+    </div>
+  </div>
+  
+  <div ng-if="newSecret.type === 'source'">
+    <div class="form-group">
+      <div class="checkbox">
+        <label>
+          <input type="checkbox" ng-model="addGitconfig">
+          Use a custom .gitconfig file
+        </label>
+        <div class="help-block">
+          Content of your .gitconfig file to authenticate with the Git server.
```

Comment:
Isn't this help text for the checkbox and not for the file input below? I'd probably just remove the help block unless we can tell the user when they need to provide a .gitconfig.